lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -206,9 +215,20 @@ def _request_certificate_renewal(
         self, scope: Scope, cert_type: CertType, secrets: Dict[str, str]
     ):
         """Request new certificate and store the key/key-password/csr in the scope's data bag."""
-        key = secrets["key"].encode("utf-8")
+        key = secrets.get("key")
+        if not key:
+            logger.error("Key %s key not found")
+            return
+
+        key = key.encode("utf-8")
         key_password = secrets.get("key-password", None)
-        old_csr = secrets["csr"].encode("utf-8")
+
+        old_csr = secrets.get("csr")
+        if not old_csr:
```

Comment:
We should not return in case the key or csr are not present, we should let it crash

Reply:
According to the old code, if either was defined as `None`, we carried on with the execution.
Technically.
(I don't know if this scenario ever happened, even on failure.)

Now the new code doesn't allow for `None` -- instead it removes the key from the secret.
I thought to unify the behavior regardless if the code is executed on Juju2 and Juju3.

Which I still think we should do: what is the preferred behavior then?
Should I raise an exception instead of returning?

Reply:
Do nothing